### PR TITLE
SAPI - Language codes and other fields to not be Empty (fix for #384)

### PR DESCRIPTION
**Pull request by willwade**
nb; using locale to get correct lang codes that are understandable

This matches eSpeak too 

You should now get data like

```
<Voice id=HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices\Tokens\TTS_MS_EN-US_DAVID_11.0
          name=Microsoft David Desktop - English (United States)
          languages=['en-US']
          gender=Male
          age=Adult>
<Voice id=HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices\Tokens\TTS_MS_EN-US_ZIRA_11.0
          name=Microsoft Zira Desktop - English (United States)
          languages=['en-US']
          gender=Female
          age=Adult>
<Voice id=HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices\Tokens\TTS_MS_ES-ES_HELENA_11.0
          name=Microsoft Helena Desktop - Spanish (Spain)
          languages=['es-ES']
          gender=Female
          age=Adult>
<Voice id=HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices\Tokens\TTS_MS_FR-FR_HORTENSE_11.0
          name=Microsoft Hortense Desktop - French
          languages=['fr-FR']
          gender=Female
          age=Adult>
<Voice id=HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Speech\Voices\Tokens\TTS_MS_RU-RU_IRINA_11.0
          name=Microsoft Irina Desktop - Russian
          languages=['ru-RU']
          gender=Female
          age=Adult>
```

### CODE REVIEW

**Comment by willwade** [comment]
RE: #385 @soleyhyman - take a look if you can

**Comment by willwade** [comment]
nb: @cclauss i see now something you were trying to tell me about yesterday - its timing out on espeak get voices in the test,

File: pyttsx3/drivers/sapi5.py [hunk 1]
```diff
@@ -88,7 +98,40 @@ def save_to_file(self, text, filename):
...
+
+        # Retrieve gender
+        gender_attr = attr.GetAttribute("Gender")
+        gender = (
+            "Male"
+            if gender_attr.lower() == "male"
+            else "Female"
+            if gender_attr.lower() == "female"
+            else None
+        )
```
Comment (cclauss) [inline pyttsx3/drivers/sapi5.py:119]:
```suggestion
        gender_title_case = gender.title()
        gender = gender_title_case if gender_title_case in {"Male", "Female"} else None
```

Comment (willwade) [reply pyttsx3/drivers/sapi5.py:119]: That was ruff making it like that. I found it wierd myself.

File: pyttsx3/drivers/sapi5.py [hunk 2]
```diff
@@ -34,6 +35,15 @@ def buildDriver(proxy):
     return SAPI5Driver(proxy)
 
 
+def lcid_to_locale(language_code):
+    primary, sub = language_code.split("-")
+    lcid = (int(sub) << 10) | int(primary)
+    try:
+        return locale.windows_locale[lcid].replace("_", "-")
+    except KeyError:
+        return "Unknown Locale"
```
[the remark this answers is not in the document]
Comment (willwade) [reply pyttsx3/drivers/sapi5.py:44]: > Where is windows_locale defined?  https://docs.python.org/3/library/locale.html
> 
> 
> 
> ```suggestion
> 
>     locale_id = (int(sub) << 10) | int(primary)
> 
>     try:
> 
>         return locale.windows_locale[locale_id].replace("_", "-")
> 
>     except KeyError:
> 
>         return f"Unknown Locale: {locale_id}"
> 
> ```

Locale. I've added it as an import right?

File: pyttsx3/drivers/sapi5.py [hunk 3]
```diff
@@ -88,7 +98,35 @@ def save_to_file(self, text, filename):
...
+
+        # Retrieve and convert language code
+        language_attr = attr.GetAttribute("Language")
+        language_code = int(language_attr, 16)
+        primary_sub_code = f"{language_code & 0x3FF}-{(language_code >> 10) & 0x3FF}"
+        languages = [lcid_to_locale(primary_sub_code)]
+
+        # Retrieve gender
+        gender_attr = attr.GetAttribute("Gender")
+        gender_title_case = (gender or "").title()
```
Comment (cclauss) [inline pyttsx3/drivers/sapi5.py:113]:
```suggestion
        gender_title_case = (gender_attr or "").title()
```

File: pyttsx3/drivers/sapi5.py [hunk 4]
```diff
@@ -88,7 +98,35 @@ def save_to_file(self, text, filename):
...
+
+        # Retrieve age
+        age_attr = attr.GetAttribute("Age")
+        age_map = {
+            "Child": "Child",
+            "Teen": "Teen",
+            "Adult": "Adult",
+            "Senior": "Senior",
+        }
+        age = age_map.get(age_attr, None)
```
Comment (cclauss) [inline pyttsx3/drivers/sapi5.py:124]:
```suggestion
        age = age_attr if age_attr in {"Child", "Teen", "Adult", "Senior"} else None
```

**Pull request merged**